go/libraries/doltcore/sqle/dprocedures/dolt_branch.go

```diff
@@ -144,7 +144,13 @@ func renameBranch(ctx *sql.Context, dbData env.DbData, apr *argparser.ArgParseRe
 		return err
 	}
 
-	err := actions.RenameBranch(ctx, dbData, oldBranchName, newBranchName, sess.Provider(), force, rsc)
+	headRef, err := dbData.Rsr.CWBHeadRef()
```

Comment:
This is kind of subtle... in addition to the current head ref, you also need to make sure to check the current checked out branch, which is what is returned by `DoltSession.CurrentHead()`. `CWBHeadRef()` will be different from the checked out branch in some cases, eg. after a `USE mydb/branch` statement. Your goal is that after this procedure is called, there are no remaining references to the old name in the session state.

Check out `DoltSession.RemoveBranchState()`, which is used when we delete a branch. We need something similar to use for this case, something like `DoltSession.RenameBranchState()`. Instead of checking out the default branch like `RemoveBranchState`, the new method should rename any references as appropriate.

Reply:
Ah, I didn't realize that `DoltSession.CurrentHead()` and `CWBHeadRef()` could diverge.

The potential issue here is that when I don't commit the transaction, `DoltSession.lookupDbState` fails when it calls `addDb`: it can't add the initial state to the session because it can't find the new branch. I imagine this is because the branch was added by directly manipulating the DB after the transaction has already been started, and the function uses the initial DB to fetch the branch state.

So I could do this, but unless I rewrite how this creates the new branch only use DoltSession (which is doable but would be a learning spike), I'd still need to manually commit the transaction first.

Reply:
Oh wait, I see what you're suggesting.

Reply:
How's this?

Reply:
LGTM, thanks